garcon/decider.py

```diff
@@ -196,7 +197,20 @@ def run(self):
                 running process.
         """
 
-        poll = self.poll()
+        try:
+            poll = self.poll()
+        except Exception as error:
+            # Catch exceptions raised during poll() to avoid a Decider thread
+            # dying & the daemon unable to process subsequent workflows.
+            # on_exception() can be overriden by the flow to send an alert
+            # when such an exception occurs.
+            if self.on_exception:
+                self.on_exception(self, error)
+            else:
+                traceback.print_stack()
+                print('Exception: {}'.format(error))
```

Comment:
A logger should be added to the decider too. Two possibilities: 
1/ have a pr that adds the logger. Update this pr to replace the print statement. (preferred)
2/ remove the print statement, and add a todo